src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/OrderingHelper.cs

```diff
@@ -14,6 +15,51 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Input.Commands.Ordering
     /// </summary>
     internal static class OrderingHelper
     {
+        /// <summary>
+        /// Performs a move on any items that were added based on the previous includes.
+        /// </summary>
+        public static Task Move(ConfiguredProject configuredProject, IProjectAccessor accessor, ImmutableHashSet<string> previousIncludes, IProjectTree target, OrderingMoveAction action)
+        {
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+            Requires.NotNull(accessor, nameof(accessor));
+            Requires.NotNull(previousIncludes, nameof(previousIncludes));
+            Requires.NotNull(target, nameof(target));
+
+            return accessor.OpenProjectForWriteAsync(configuredProject, project =>
+            {
+                // We do a sanity re-evaluation to absolutely ensure changes were met.
+                project.ReevaluateIfNecessary();
```

Comment:
I don't understand this - and is this legal inside a lock @lifengl?

Reply:
Ping @Lifengl - do you know if this is safe to do while we hold the project lock?

Reply:
It looks legal inside a lock (I missed that project was MSBuid's project). If it is needed, I'd recommend updating the comment indicating why exactly.

Reply:
It was really just for sanity to be absolutely sure changes are made; I can't recall any bugs without it though. Technically we might be able to get rid of it.